src/env.cc

```diff
@@ -847,14 +847,12 @@ Environment::Environment(IsolateData* isolate_data,
     }
   }
 
-  // We are supposed to call builtin_loader_.SetEagerCompile() in
-  // snapshot mode here because it's beneficial to compile built-ins
-  // loaded in the snapshot eagerly and include the code of inner functions
-  // that are likely to be used by user since they are part of the core
-  // startup. But this requires us to start the coverage collections
-  // before Environment/Context creation which is not currently possible.
-  // TODO(joyeecheung): refactor V8ProfilerConnection classes to parse
-  // JSON without v8 and lift this restriction.
+  // Compile builtins eagerly when building the snapshot so that inner functions
+  // of essential builtins that are loaded in the snapshot can have faster first
+  // invocation.
+  if (isolate_data->is_building_snapshot()) {
+    builtin_loader()->SetEagerCompile();
```

Comment:
er, I meant to open a separate PR about this (this was why I opened https://github.com/nodejs/node/pull/54633 which remove the coverage regression I initially discovered in https://github.com/nodejs/node/pull/51672), somehow this ended up in the checkout of this PR due to local git mess up 😅 but it was approved and landed anyway..

Reply:
For posterity: the short summary of this small diff that accidentally get landed in this unrelated PR is that it speeds up core startup by not having to compile any inner functions in essential builtins. Previously, only the top-level functions are cached in the snapshot, so during bootstrap, when the inner functions are executed to do the loading, they still need to be compiled without cache. This diff changed to cache the inner functions too, which adds ~630KB to the binary, in exchange of a slightly faster core startup (and also speeding up the first invocation of most functions provided by the essential builtins).

❯ hyperfine "./node_pr ./test/fixtures/semicolon.js" "./node_main ./test/fixtures/semicolon.js"
Benchmark 1: ./node_pr ./test/fixtures/semicolon.js
  Time (mean ± σ):      34.2 ms ±   1.5 ms    [User: 30.1 ms, System: 2.5 ms]
  Range (min … max):    33.1 ms …  43.7 ms    80 runs

Benchmark 2: ./node_main ./test/fixtures/semicolon.js
  Time (mean ± σ):      35.1 ms ±   1.2 ms    [User: 31.0 ms, System: 2.5 ms]
  Range (min … max):    34.1 ms …  42.6 ms    77 runs

Summary
  './node_pr ./test/fixtures/semicolon.js' ran
    1.03 ± 0.06 times faster than './node_main ./test/fixtures/semicolon.js'